Gemfile

```diff
@@ -42,14 +42,14 @@ supports_windows = true
 
 group :development do
   gem 'puppet-lint',                        :require => false
-  gem 'metadata-json-lint',                 :require => false if ! supports_windows || Gem::Version.new(RUBY_VERSION.dup) >= Gem::Version.new('2.3.1')
+  gem 'metadata-json-lint',                 :require => false, :platforms => 'ruby'
```

Comment:
I'm guessing it does all of the resolving without regard to what it is actually pulling down because that path is easier and the gems and versions that you could ultimately get would not have conflicts, etc. Especially if you changed your mind and decided to include everything. Lots of likely reasons why it resolves all gems and dependencies, but the biggest is it makes things more consistent and a simpler path. But again, even though it resolves json for beaker, it doesn't install either when the gem is in the `--without` section. 

If you are seeing otherwise, it is not because of beaker (or if it is, it's a bug in the bundler version).